spec/system/admin/subscriptions/smoke_tests_spec.rb

```diff
@@ -242,15 +242,21 @@
         expect(page).to have_selector "#subscription-line-items .item", count: 4
 
         # Delete an existing product
-        puts "TODO: migrate to v3" # but first we need to make the actions menu more accessible.
-        Flipper.disable(:admin_style_v3) # disabling BUU for legacy products page
-        login_as_admin
-        visit spree.admin_products_path
-        within "#p_#{shop_product2.id}" do
-          accept_alert { page.find("[data-powertip=Remove]").click }
+        visit admin_products_url
+
+        product_selector = row_containing_name(shop_product2.name)
+        delete_option_selector = "a[data-controller='modal-link'].delete"
+        delete_button_selector = "input[type=button][value='Delete product']"
+        modal_selector = "div[data-modal-target=modal]"
+
+        within product_selector do
+          page.find(".vertical-ellipsis-menu").click
+          page.find(delete_option_selector).click
+        end
+        within modal_selector do
+          page.find(delete_button_selector).click
```

Comment:
And `click_button`?

Reply:
The confirm/cancel modal has is indeed elements which are `type=button`:

`<input class="button icon-plus red" data-action="click->products#delete_product click->modal#close" id="modal-confirm-button" type="button" value="Delete product" data-delete-path="/admin/products_v3/22">`

So, `click_button "Delete product"` works :smile:

Reply:
@dacook I went ahead and applied your proposed improvement on the the [v3 products spec](https://github.com/openfoodfoundation/openfoodnetwork/pull/12652/commits/4e5ab0f77fd92510b34bf8062de929db569e9169).